src/main/java/com/salesforce/dataloader/security/EncryptionAesUtil.java

```diff
@@ -0,0 +1,255 @@
+            if (!Files.exists(Paths.get(filePath))) {
+                byte[] key = generateEncryptionKey();
+                try (FileOutputStream fos = new FileOutputStream(filePath)) {
+                    fos.write(key);
+                } catch (IOException io) {
+                    throw new GeneralSecurityException("Failed to open file:" + filePath, io);
+                }
+                // Windows platform is already readable only to owner.
+                if (detectedOS == OSType.MacOS || detectedOS == OSType.Linux) {
+                    File file = new File(filePath);
+                    // set all reading to false
+                    file.setReadable(false, false);
+                    // set Owner reading to true
+                    file.setReadable(true, true);
+                }
+            }
+        }
+        setCipherKeyFromFilePath(filePath);
+        return filePath;
+    }
+
+    public void resetCipherKey() {
+        cipherKey = null;
+    }
+
+    public void setCipherKeyFromFilePath(String filePath) throws GeneralSecurityException {
+        byte[] data = new byte[1024];
+        try (FileInputStream fis = new FileInputStream(filePath)) {
+            int size = fis.read(data);
+            if (size < ENCRYPTION_KEY_LENGTH_IN_BYTES)
+                throw new GeneralSecurityException("Keyfile content is too short:" + filePath);
+            else {
+                cipherKey = new byte[ENCRYPTION_KEY_LENGTH_IN_BYTES];
+                System.arraycopy(data, 0, cipherKey, 0, ENCRYPTION_KEY_LENGTH_IN_BYTES);
+            }
+        } catch (IOException io) {
+            throw new GeneralSecurityException("Failed to open file: " + filePath, io);
+        }
+    }
+
+    private void ensureKeyIsSet() throws IOException, GeneralSecurityException {
+        if (cipherKey != null) {
+            return;
+        }
+        // if no key was set up already, create one
+        createKeyFileIfNotExisting(null);
+
+    }
+
+    public String encryptMsg(String msg) throws GeneralSecurityException {
+        try {
+            ensureKeyIsSet();
+            return EncryptionUtil.bytesToText(encryptMsg(msg, cipherKey));
+        } catch (Exception e) {
+            LOGGER.error("Fail to encrypt message: " + e.getMessage());
+            throw new GeneralSecurityException("Error to encrypt message: ", e);
+        }
+    }
+
+    public String decryptMsg(String cipherMsgString) throws GeneralSecurityException {
+        try {
+            ensureKeyIsSet();
+            byte[] cipherMsg = EncryptionUtil.textToBytes(cipherMsgString);
+            return decryptMsg(cipherMsg, cipherKey);
+        } catch (Exception e) {
+            LOGGER.error("Fail to decrypt message: " + e.getMessage());
+            throw new GeneralSecurityException("Error to encrypt message: ", e);
+        }
+    }
+
+    synchronized public byte[] encryptMsg(String msg, byte[] encryptionKey) throws GeneralSecurityException, UnsupportedEncodingException {
```

Comment:
Why is this function `synchronized` and `public`? It's not `static`, so it synchronizes against the instance, but you also said thread safety isn't a goal here in which case synchronizing at all is just misleading (and also wasteful of performance). It's also not clear why both these encrypt/decrypt functions and the versions that don't take an explicit key parameter are `public`.